src/Core/Resolvers/SqlQueryEngine.cs

```diff
@@ -170,32 +172,68 @@ public async Task<IActionResult> ExecuteAsync(StoredProcedureRequestContext cont
         }
 
         /// <inheritdoc />
-        public JsonDocument? ResolveInnerObject(JsonElement element, IObjectField fieldSchema, ref IMetadata metadata)
+        public JsonElement ResolveObject(JsonElement element, IObjectField fieldSchema, ref IMetadata metadata)
         {
             PaginationMetadata parentMetadata = (PaginationMetadata)metadata;
+            if (parentMetadata.Subqueries.TryGetValue(QueryBuilder.PAGINATION_FIELD_NAME, out PaginationMetadata? paginationObjectMetadata))
+            {
+                parentMetadata = paginationObjectMetadata;
+            }
+
             PaginationMetadata currentMetadata = parentMetadata.Subqueries[fieldSchema.Name.Value];
             metadata = currentMetadata;
 
             if (currentMetadata.IsPaginated)
             {
                 return SqlPaginationUtil.CreatePaginationConnectionFromJsonElement(element, currentMetadata);
             }
-            else
-            {
-                //TODO: Try to avoid additional deserialization/serialization here.
-                return ResolverMiddleware.RepresentsNullValue(element) ? null : JsonDocument.Parse(element.ToString());
-            }
+
+            return element;
         }
 
-        /// <inheritdoc />
-        public object? ResolveListType(JsonElement element, IObjectField fieldSchema, ref IMetadata metadata)
+        /// <summary>
+        /// Resolves the JsonElement, an array, into a list of jsonelements where each element represents
+        /// an entry in the original array.
+        /// </summary>
+        /// <param name="array">JsonElement representing a JSON array. The possible representations:
+        /// JsonValueKind.Array -> ["item1","itemN"]
+        /// JsonValueKind.String (array of objects) -> [ { "field1": "field1Value", "field2": "field2Value" }, { ... } ]
```

Comment:
It makes more sense that array of objects would be `JsonValueKind.Array` just like the first example. But it seems that however the JsonElement is created or how System.Text.Json resolves the object, it is determining it to be a string.

Reply:
would the sample value be `"[ { "field1": "field1Value", "field2": "field2Value" }, { ... } ]"` instead?

Reply:
Yep, thanks for catching the missing `}`

Also updated the comment to specifically call out why the "JsonValueKind.String" is stamped on the JsonElement when the contents imply that the valuekind should be array. Essentially, it's because the contents are serialized JSON and haven't been parsed/deserialized yet.  
```csharp
        /// <param name="array">JsonElement representing a JSON array. The possible representations:
        /// JsonValueKind.Array -> ["item1","itemN"]
        /// JsonValueKind.String -> "[ { "field1": "field1Value" }, { "field2": "field2Value" }, { ... } ]"
        /// - Input JsonElement is JsonValueKind.String because the array and enclosed objects haven't been parsed yet.
        /// - This method parses the JSON string (representing a JSON array) and collects each element (Json object) within the
        /// list of json elements returned by this method.</param>
```

Contents in debugger:
![image](https://github.com/Azure/data-api-builder/assets/6414189/0dcb41ff-2dbc-4a6e-9652-f9948563791a)